src/overlay/OverlayManagerImpl.cpp

```diff
@@ -192,29 +192,51 @@ OverlayManagerImpl::storeConfigPeers()
 }
 
 void
-OverlayManagerImpl::connectToMorePeers(int max)
+OverlayManagerImpl::connectToMorePeers(int max, bool preferred)
```

Comment:
I would prefer this to look more like:

```
connectToPeers(vector<peer>);
vector<peer> getPrefferedPeersToConnect(int max)
vector<peer> getPeersToConnect(int max);
```

With calls like:

```
connectToPeers(getPrefferedPeersToConnect(max));
connectToPeers(getPeersToConnect(max));
```

Instead of this bool flag